client/src/includes/userbar.ts

```diff
@@ -0,0 +1,458 @@
+import axe, { AxeResults } from 'axe-core';
+
+import { dialog } from './dialog';
+
+/*
+This entrypoint is not bundled with any polyfills to keep it as light as possible
+Please stick to old JS APIs and avoid importing anything that might require a vendored module
+More background can be found in webpack.config.js
+
+This component implements a roving tab index for keyboard navigation
+Learn more about roving tabIndex: https://w3c.github.io/aria-practices/#kbd_roving_tabindex
+*/
+
+export class Userbar extends HTMLElement {
+  declare trigger: HTMLElement;
+
+  connectedCallback() {
+    const template = document.querySelector<HTMLTemplateElement>(
+      '#wagtail-userbar-template',
+    );
+    if (!template) return;
+    const shadowRoot = this.attachShadow({
+      mode: 'open',
+    });
+    shadowRoot.appendChild(template.content.cloneNode(true));
+    // Removes the template from html after it's being used
+    template.remove();
+
+    const userbar = shadowRoot.querySelector<HTMLDivElement>(
+      '[data-wagtail-userbar]',
+    );
+    if (!userbar) return;
+    const trigger = userbar.querySelector<HTMLElement>(
+      '[data-wagtail-userbar-trigger]',
+    );
+    if (!trigger) return;
+    this.trigger = trigger;
+    const list = userbar.querySelector('[role=menu]');
+    if (!list) return;
+    const listItems = list.querySelectorAll('li');
+    const isActiveClass = 'is-active';
+
+    // Avoid Web Component FOUC while stylesheets are loading.
+    userbar.style.display = 'none';
+
+    /*
+    querySelector for all items that can be focused
+    tabIndex has been removed for roving tabindex compatibility
+    source: https://stackoverflow.com/questions/1599660/which-html-elements-can-receive-focus
+    */
+    const focusableItemSelector = `a[href],
+    button:not([disabled]),
+    input:not([disabled])`;
+
+    const showUserbar = (shouldFocus) => {
+      userbar.classList.add(isActiveClass);
+      trigger.setAttribute('aria-expanded', 'true');
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      list.addEventListener('click', sandboxClick, false);
+      // eslint-disable-next-line @typescript-eslint/no-use-before-define
+      window.addEventListener('click', clickOutside, false);
```

Comment:
I'm surprised that these require the eslint-disable directives - is there another way to solve this?

Reply:
I agree, perhaps it's possible to either move this `showUserbar` function after those handlers' definitions